### ISSUE TITLE
Add property on RawTCPLayer to indicate which end closed the connection

### ISSUE THREAD

**Issue by Caligatio** [mitmproxy/mitmproxy#2408]
Following up on my other issue, #2382, it would be helpful to add a property/flag to the `RawTCPLayer` that indicates whether the client or server closed the connection.  The main use case for me is using the addon API and receiving `tcp_start`, `tcp_message`, and `tcp_end` events.  The initiator on `tcp_start` is obvious (the client), the sender on `tcp_message` is given in the `TCPMessage` object, but no similar information is exposed to the `tcp_end` event.

I'm not super confident but I think [Line 47 in rawtcp.py](/mitmproxy/mitmproxy/blob/master/mitmproxy/proxy/protocol/rawtcp.py#L47) would be the right spot to add something like this.

**Comment by mhils** [comment]
Makes sense. `tcp_end` is currently emitted when we are sure that neither side is sending (expected) messages anymore. If we want to notify users that one half has closed, maybe we can add a `tcp_close` event at [rawtcp.py#L40](https://github.com/mitmproxy/mitmproxy/blob/bde64746a91666a3ae8b693fed05bc29824b814c/mitmproxy/proxy/protocol/rawtcp.py#L40). 

refs #1775

**Comment by Kriechi** [comment]
@mhils still relevant with the new proxy core?

**Comment by mhils** [comment]
Yes, we still don't expose this.
